pkg/resources/management.cattle.io/v3/projectroletemplatebinding/validator.go

```diff
@@ -198,9 +204,38 @@ func (a *admitter) validateCreateFields(newPRTB *apisv3.ProjectRoleTemplateBindi
 		return field.NotSupported(fieldPath.Child("roleTemplate", "context"), roleTemplate.Context, []string{projectContext})
 	}
 
-	_, projectNS := clusterAndProjectID(newPRTB.ProjectName)
-	if projectNS != newPRTB.Namespace {
-		return field.Forbidden(fieldPath, "namespace and the project ID part of projectName must match")
+	clusterName, projectName := clusterAndProjectID(newPRTB.ProjectName)
+	if clusterName == "" || projectName == "" {
+		return field.Required(fieldPath.Child("projectName"), "projectName must be of the form clusterName:projectName, and both must refer to an existing object")
```

Comment:
1. We should check if newPRTB.ProjectName is empty first since that is the only time we should be using the required error. If the format is incorrect such as `:p-fscsfd` then we should instead use the` field.Invalid(...)` error.
2. `clusterName:projectName` in the message may be confusing since the field name is also `projectName`
instead, we could say `cluster.metadata.name:project.metadata.name`

Reply:
I've made the changes as requested.